core/src/org/pentaho/di/cluster/HttpUtil.java

```diff
@@ -154,11 +161,17 @@ public static int execMethod( HttpClient client, HttpMethod method ) throws Exce
    */
   public static String constructUrl( VariableSpace space, String hostname, String port, String webAppName,
     String serviceAndArguments ) throws UnsupportedEncodingException {
+    return constructUrl( space, hostname, port, webAppName, serviceAndArguments, false );
+  }
+
+  public static String constructUrl( VariableSpace space, String hostname, String port, String webAppName,
+      String serviceAndArguments, boolean isSecure ) throws UnsupportedEncodingException {
     String realHostname = space.environmentSubstitute( hostname );
     if ( !StringUtils.isEmpty( webAppName ) ) {
       serviceAndArguments = "/" + space.environmentSubstitute( webAppName ) + serviceAndArguments;
     }
-    String retval = "http://" + realHostname + getPortSpecification( space, port ) + serviceAndArguments;
+    String protocol = "http" + ( isSecure ? "s" : "" );
+    String retval = protocol + "://" + realHostname + getPortSpecification( space, port ) + serviceAndArguments;
```

Comment:
I would have preferred to see 2 string constants (UNSECURE_PROTOCOL, SECURE_PROTOCOL) instead of building the string right in the middle of this code.